app/models/claim.rb

```diff
@@ -140,6 +141,22 @@ def payroll_gender_missing?
     %w[male female].exclude?(payroll_gender)
   end
 
+  def awaiting_check?
+    submitted? && check.nil?
+  end
+
+  def approaching_deadline?
+    awaiting_check? && Time.zone.now > (deadline_date - 2.weeks) && !passed_deadline?
+  end
+
+  def passed_deadline?
+    awaiting_check? && Time.zone.now > deadline_date
+  end
+
+  def deadline_date
```

Comment:
As with the constant above, would be good to give these methods more specific names, e.g. `checking_deadline_date`